vhdbuilder/packer/init-variables.sh

```diff
@@ -419,7 +419,9 @@ cat <<EOF > vhdbuilder/packer/settings.json
   "windows_sigmode_source_resource_group_name": "${windows_sigmode_source_resource_group_name}",
   "windows_sigmode_source_gallery_name": "${windows_sigmode_source_gallery_name}",
   "windows_sigmode_source_image_name": "${windows_sigmode_source_image_name}",
-  "windows_sigmode_source_image_version": "${windows_sigmode_source_image_version}"
+  "windows_sigmode_source_image_version": "${windows_sigmode_source_image_version}",
+  "vnet_name": "nodesig-pool-vnet",
```

Comment:
did you test windows? if so nice, otherwise would leave this alone until @AbelHu or someone else can take a look. I hope it "just works" but not 100% sure how the network topology looks there.

Reply:
haven't tested on windows as of yet

Reply:
@cameronmeissner could you help to introduce what this PR is for?
When building Windows VHDs,
- we have used a pre-created vnet: https://github.com/Azure/AgentBaker/blob/master/vhdbuilder/packer/init-variables.sh#L39-L66
- we have rebooted the build VM: https://github.com/Azure/AgentBaker/blob/master/vhdbuilder/packer/windows-vhd-builder.json#L76

Reply:
this is for 1es build pools. we pre-create vnet with **no inbound connectivity**. this fixes our security exception!

the 1es build agents are in the vnet Cameron specified here. it doesn't look like windows nsg has anything special, just allow WinRM SSL? that should work here, but if you're already pre-creating a vnet, likely we could refactor a bit more.

Reply:
Yes. we pre-create the vnet for Windows VHD build VM since we need to add the tag `SkipASMAzSecPack=True` and allow WinRM with SSL 5986. It seems like that Windows also can reuse this PR to avoid the security exception if we use the vnet Cameron specified here and 1es build pools. Please free to correct me if my understanding is wrong

Reply:
can we create nodesig-pool-vnet dynamically? Is it ok to use the same vent?

Reply:
do you mean dynamically creating nodesig-pool-vnet on demand or something? pretty sure as long as there isn't already a vnet with the same name in the target resource group it should be fine. although, I think a simpler approach would be to just modify the bicep templates we're using for deploying the 1es pools to add another subnet within nodesig-pool-vnet that we can then point to from the packer templates in a similar fashion as we're doing now for linux.